src/table/table-adapter.class.ts

```diff
@@ -181,4 +263,66 @@ export class TableDomAdapter implements TableAdapter {
 	findIndex(cell: HTMLTableCellElement): [number, number] {
 		return [this.findRowIndex(cell), this.findColumnIndex(cell)];
 	}
+
+	/**
+	 * Helper function that returns the "real" length of a row.
+	 * Only accurate with regard to colspans (though that's sufficient for it's uses here)
+	 *
+	 * TODO: Take rowSpan into account
+	 *
+	 * @param row the row to get the length of
+	 */
+	protected getRealRowLength(row: HTMLTableRowElement): number {
+		return Array.from(row.cells).reduce((count, cell) => {
+			return count + cell.colSpan;
+		}, -1); // start at -1 since the colspans will sum to 1 index greater than the total
+	}
+
+	/**
+	 * Finds a cell and it's real index given an array of cells, a target index, and the spanning direction
+	 *
+	 * @param cells An array of cells to search
+	 * @param targetIndex The index we think the cell is located at
+	 * @param spanDirection The direction of the cell spans. Should be `"colSpan"` for a row and `"rowSpan"` for a column
+	 */
+	protected findCell(cells: HTMLTableCellElement[], targetIndex: number, spanDirection: "colSpan" | "rowSpan") {
```

Comment:
Can `spanDirection` be an enum?

Reply:
Maybe! It kinda is already, but an explicit enum may help